<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Package index • SNPtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Introduction.html">Introduction</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/viniciusjunqueira/SNPtools/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="FImputeExport-class.html">FImputeExport-class</a></code> </p>
        </td>
        <td><p>FImputeExport Class</p></td>
      </tr><tr><td>
          <p><code><a href="FImputeRunner-class.html">FImputeRunner-class</a></code> </p>
        </td>
        <td><p>FImputeRunner Class</p></td>
      </tr><tr><td>
          <p><code><a href="FImputeRunner.html">FImputeRunner()</a></code> </p>
        </td>
        <td><p>Build FImputeRunner object</p></td>
      </tr><tr><td>
          <p><code><a href="SNPDataLong-class.html">SNPDataLong-class</a></code> </p>
        </td>
        <td><p>SNPDataLong Class</p></td>
      </tr><tr><td>
          <p><code><a href="SNPFileConfig-class.html">SNPFileConfig-class</a></code> </p>
        </td>
        <td><p>SNPFileConfig Class</p></td>
      </tr><tr><td>
          <p><code><a href="SNPImportList-class.html">SNPImportList-class</a></code> </p>
        </td>
        <td><p>SNPImportList Class</p></td>
      </tr><tr><td>
          <p><code><a href="Subset.html">Subset()</a></code> </p>
        </td>
        <td><p>Subset method for SNPDataLong</p></td>
      </tr><tr><td>
          <p><code><a href="cbind_SnpMatrix.html">cbind_SnpMatrix()</a></code> </p>
        </td>
        <td><p>Safe cbind for SnpMatrix preserving dimnames</p></td>
      </tr><tr><td>
          <p><code><a href="check.call.rate.html">check.call.rate()</a></code> </p>
        </td>
        <td><p>Check SNP call rate</p></td>
      </tr><tr><td>
          <p><code><a href="check.ibs.html">check.ibs()</a></code> </p>
        </td>
        <td><p>Check Identity-By-State (IBS) for a genotype pair</p></td>
      </tr><tr><td>
          <p><code><a href="check.identical.samples.html">check.identical.samples()</a></code> </p>
        </td>
        <td><p>Check identical samples based on distance</p></td>
      </tr><tr><td>
          <p><code><a href="check.identical.samples.by.block.html">check.identical.samples.by.block()</a></code> </p>
        </td>
        <td><p>Check identical samples by block</p></td>
      </tr><tr><td>
          <p><code><a href="check.mendelian.inconsistencies.html">check.mendelian.inconsistencies()</a></code> </p>
        </td>
        <td><p>Check Mendelian inconsistencies</p></td>
      </tr><tr><td>
          <p><code><a href="check.mendelian.inconsistencies.pair.html">check.mendelian.inconsistencies.pair()</a></code> </p>
        </td>
        <td><p>Check Mendelian inconsistencies for a pair</p></td>
      </tr><tr><td>
          <p><code><a href="check.sample.call.rate.html">check.sample.call.rate()</a></code> </p>
        </td>
        <td><p>Check Sample Call Rate</p></td>
      </tr><tr><td>
          <p><code><a href="check.sample.heterozygosity.html">check.sample.heterozygosity()</a></code> </p>
        </td>
        <td><p>Check sample heterozygosity</p></td>
      </tr><tr><td>
          <p><code><a href="check.snp.chromo.html">check.snp.chromo()</a></code> </p>
        </td>
        <td><p>Check SNP by chromosome</p></td>
      </tr><tr><td>
          <p><code><a href="check.snp.hwe.html">check.snp.hwe()</a></code> </p>
        </td>
        <td><p>Check SNP Hardy-Weinberg equilibrium deviation</p></td>
      </tr><tr><td>
          <p><code><a href="check.snp.hwe.chi2.html">check.snp.hwe.chi2()</a></code> </p>
        </td>
        <td><p>Check SNPs for Hardy-Weinberg equilibrium deviation using chi-square p-values</p></td>
      </tr><tr><td>
          <p><code><a href="check.snp.maf.html">check.snp.maf()</a></code> </p>
        </td>
        <td><p>Check SNP minor allele frequency</p></td>
      </tr><tr><td>
          <p><code><a href="check.snp.mgf.html">check.snp.mgf()</a></code> </p>
        </td>
        <td><p>Check SNP missing genotype frequencies</p></td>
      </tr><tr><td>
          <p><code><a href="check.snp.monomorf.html">check.snp.monomorf()</a></code> </p>
        </td>
        <td><p>Check SNP monomorphic status</p></td>
      </tr><tr><td>
          <p><code><a href="check.snp.no.position.html">check.snp.no.position()</a></code> </p>
        </td>
        <td><p>Check SNP no position</p></td>
      </tr><tr><td>
          <p><code><a href="check.snp.same.position.html">check.snp.same.position()</a></code> </p>
        </td>
        <td><p>Check SNPs mapped to the same position</p></td>
      </tr><tr><td>
          <p><code><a href="combinarSNPData.html">combinarSNPData()</a></code> </p>
        </td>
        <td><p>Combine multiple SNPDataLong objects</p></td>
      </tr><tr><td>
          <p><code><a href="doPCA.html">doPCA()</a></code> </p>
        </td>
        <td><p>Do genome relationship matrix PCA</p></td>
      </tr><tr><td>
          <p><code><a href="exploratory.plots.html">exploratory.plots()</a></code> </p>
        </td>
        <td><p>Exploratory plots for SNP and sample summary</p></td>
      </tr><tr><td>
          <p><code><a href="genoToDF.html">genoToDF()</a></code> </p>
        </td>
        <td><p>Convert geno slot from SNPDataLong to a data.frame</p></td>
      </tr><tr><td>
          <p><code><a href="get.correl.fc.html">get.correl.fc()</a></code> </p>
        </td>
        <td><p>Get correlation (fc method)</p></td>
      </tr><tr><td>
          <p><code><a href="get.gender.html">get.gender()</a></code> </p>
        </td>
        <td><p>Get gender based on heterozygosity</p></td>
      </tr><tr><td>
          <p><code><a href="get.hwe.chi2.html">get.hwe.chi2()</a></code> </p>
        </td>
        <td><p>Get HWE chi-square p-values</p></td>
      </tr><tr><td>
          <p><code><a href="getGeno.html">getGeno()</a></code> </p>
        </td>
        <td><p>Flexible and efficient genotype file reading with autodetection using fread</p></td>
      </tr><tr><td>
          <p><code><a href="ibs.pair.html">ibs.pair()</a></code> </p>
        </td>
        <td><p>IBS pair statistics</p></td>
      </tr><tr><td>
          <p><code><a href="importAllGenos.html">importAllGenos()</a></code> </p>
        </td>
        <td><p>Import and combine multiple genotype configurations</p></td>
      </tr><tr><td>
          <p><code><a href="importFImputeResults.html">importFImputeResults()</a></code> </p>
        </td>
        <td><p>Import imputed FImpute results from disk</p></td>
      </tr><tr><td>
          <p><code><a href="import_geno_list.html">import_geno_list()</a></code> </p>
        </td>
        <td><p>Import multiple genotype datasets from a list of configurations</p></td>
      </tr><tr><td>
          <p><code><a href="pairs2sets.html">pairs2sets()</a></code> </p>
        </td>
        <td><p>Convert pairs to sets</p></td>
      </tr><tr><td>
          <p><code><a href="plotPCAgroups.html">plotPCAgroups()</a></code> </p>
        </td>
        <td><p>Plot PCA groups from anticlustering result</p></td>
      </tr><tr><td>
          <p><code><a href="qcSNPs.html">qcSNPs()</a></code> </p>
        </td>
        <td><p>Quality Control for SNPDataLong with optional criteria</p></td>
      </tr><tr><td>
          <p><code><a href="qcSamples.html">qcSamples()</a></code> </p>
        </td>
        <td><p>Quality control on samples</p></td>
      </tr><tr><td>
          <p><code><a href="qc_header.html">qc_header()</a></code> </p>
        </td>
        <td><p>Formatted header message</p></td>
      </tr><tr><td>
          <p><code><a href="rbindSnpFlexible.html">rbindSnpFlexible()</a></code> </p>
        </td>
        <td><p>Faster row-bind for SnpMatrix objects with differing columns</p></td>
      </tr><tr><td>
          <p><code><a href="rbind_SnpMatrix.html">rbind_SnpMatrix()</a></code> </p>
        </td>
        <td><p>Safe rbind for SnpMatrix preserving dimnames</p></td>
      </tr><tr><td>
          <p><code><a href="read.fimpute.html">read.fimpute()</a></code> </p>
        </td>
        <td><p>Read imputed genotypes from FImpute output and return SNPDataLong object</p></td>
      </tr><tr><td>
          <p><code><a href="runAnticlusteringPCA.html">runAnticlusteringPCA()</a></code> </p>
        </td>
        <td><p>Run PCA and Anticlustering on SNPDataLong</p></td>
      </tr><tr><td>
          <p><code><a href="runFImpute.html">runFImpute()</a></code> </p>
        </td>
        <td><p>Run FImpute from a FImputeRunner object</p></td>
      </tr><tr><td>
          <p><code><a href="saveFImpute.html">saveFImpute()</a></code> </p>
        </td>
        <td><p>Save genotype and map files in FImpute format</p></td>
      </tr><tr><td>
          <p><code><a href="saveFImputeRaw.html">saveFImputeRaw()</a></code> </p>
        </td>
        <td><p>Export genotypes and map using basic arguments</p></td>
      </tr><tr><td>
          <p><code><a href="savePlink.html">savePlink()</a></code> </p>
        </td>
        <td><p>Save SNPDataLong object to PLINK format</p></td>
      </tr><tr><td>
          <p><code><a href="summary-SNPDataLong-method.html">summary(<i>&lt;SNPDataLong&gt;</i>)</a></code> </p>
        </td>
        <td><p>Summary for SNPDataLong objects</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Vinícius Junqueira.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

