<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Quality Control for SNPDataLong with optional criteria — qcSNPs • SNPtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Quality Control for SNPDataLong with optional criteria — qcSNPs"><meta property="og:description" content="Applies flexible quality control filters on an object of class SNPDataLong.
Supports call rate filtering, minor allele frequency (MAF), Hardy-Weinberg equilibrium (HWE),
removal of monomorphic SNPs, exclusion of specific chromosomes, optionally removing SNPs without positions,
and optionally removing SNPs at the same genomic position (keeping the one with highest MAF)."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Introduction.html">Introduction</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/viniciusjunqueira/SNPtools/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Quality Control for SNPDataLong with optional criteria</h1>
    <small class="dont-index">Source: <a href="https://github.com/viniciusjunqueira/SNPtools/blob/HEAD/R/qcSNPs.R" class="external-link"><code>R/qcSNPs.R</code></a></small>
    <div class="hidden name"><code>qcSNPs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Applies flexible quality control filters on an object of class <code>SNPDataLong</code>.
Supports call rate filtering, minor allele frequency (MAF), Hardy-Weinberg equilibrium (HWE),
removal of monomorphic SNPs, exclusion of specific chromosomes, optionally removing SNPs without positions,
and optionally removing SNPs at the same genomic position (keeping the one with highest MAF).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">qcSNPs</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>An object of class SNPDataLong.</p></dd>


<dt id="arg-missing-ind">missing_ind<a class="anchor" aria-label="anchor" href="#arg-missing-ind"></a></dt>
<dd><p>Maximum allowed proportion of missing data per individual (currently not implemented).</p></dd>


<dt id="arg-missing-snp">missing_snp<a class="anchor" aria-label="anchor" href="#arg-missing-snp"></a></dt>
<dd><p>Maximum allowed proportion of missing data per SNP (currently not implemented).</p></dd>


<dt id="arg-min-snp-cr">min_snp_cr<a class="anchor" aria-label="anchor" href="#arg-min-snp-cr"></a></dt>
<dd><p>Minimum acceptable call rate for SNPs (e.g., 0.95). SNPs below this threshold are removed.</p></dd>


<dt id="arg-min-maf">min_maf<a class="anchor" aria-label="anchor" href="#arg-min-maf"></a></dt>
<dd><p>Minimum minor allele frequency allowed for SNPs (e.g., 0.05). SNPs with lower MAF are removed.</p></dd>


<dt id="arg-hwe">hwe<a class="anchor" aria-label="anchor" href="#arg-hwe"></a></dt>
<dd><p>p-value threshold for Hardy-Weinberg equilibrium test (e.g., 1e-6). SNPs violating this are removed.</p></dd>


<dt id="arg-snp-position">snp_position<a class="anchor" aria-label="anchor" href="#arg-snp-position"></a></dt>
<dd><p>Logical. If TRUE, removes SNPs mapped to the same position, retaining only the one with highest MAF.</p></dd>


<dt id="arg-no-position">no_position<a class="anchor" aria-label="anchor" href="#arg-no-position"></a></dt>
<dd><p>Logical. If TRUE, removes SNPs without defined genomic positions.</p></dd>


<dt id="arg-snp-mono">snp_mono<a class="anchor" aria-label="anchor" href="#arg-snp-mono"></a></dt>
<dd><p>Logical. If TRUE, removes monomorphic SNPs (with no variation).</p></dd>


<dt id="arg-remove-chr">remove_chr<a class="anchor" aria-label="anchor" href="#arg-remove-chr"></a></dt>
<dd><p>Character vector of chromosomes to exclude (e.g., c("X", "Y")).</p></dd>


<dt id="arg-action">action<a class="anchor" aria-label="anchor" href="#arg-action"></a></dt>
<dd><p>One of "report" (returns a list of removed SNPs), "filter" (returns filtered SNPDataLong), or "both" (returns both).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Depending on the action argument:
- "report": list of SNPs removed by each filter and SNPs retained.
- "filter": filtered SNPDataLong object.
- "both": list containing the filtered object and detailed report.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="cn">NA</span><span class="op">)</span>, <span class="fl">100</span>, replace <span class="op">=</span> <span class="cn">TRUE</span>, prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.4</span>, <span class="fl">0.4</span>, <span class="fl">0.15</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>              nrow <span class="op">=</span> <span class="fl">10</span>, ncol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"snp"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"ind"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">map</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>, Chromosome <span class="op">=</span> <span class="fl">1</span>, Position <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span><span class="st">"SNPDataLong"</span>, geno <span class="op">=</span> <span class="va">mat</span>, map <span class="op">=</span> <span class="va">map</span>, path <span class="op">=</span> <span class="st">"dummy_path"</span>, xref_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"chip1"</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example using multiple filters</span></span></span>
<span class="r-in"><span><span class="fu">qcSNPs</span><span class="op">(</span><span class="va">x</span>, min_snp_cr <span class="op">=</span> <span class="fl">0.8</span>, min_maf <span class="op">=</span> <span class="fl">0.05</span>, snp_mono <span class="op">=</span> <span class="cn">TRUE</span>, no_position <span class="op">=</span> <span class="cn">TRUE</span>, snp_position <span class="op">=</span> <span class="cn">TRUE</span>, action <span class="op">=</span> <span class="st">"filter"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Vinícius Junqueira.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

